### PR TITLE
Checkpoint 3

### PR DESCRIPTION
**Pull request by javierforero**
David,

So I decided to keep it working the old way. I do want to see how this AJAX method works, though. I looked into using this gem https://github.com/bernat/best_in_place, but I am not sure I quiet understand it. Just so you know this is how I think the AJAX method would work:
1. Add an "enable" field to the RegisteredApp model setting the default to true
2. Then I would create a method in the controller called "enable" to handle the update of the "enable 
   field"
3. create an enable.js.erb file that would have an onClick handler function that would trigger the enable  
   method in the controller and also hide the respective registered app when it is clicked.

### CODE REVIEW

**Comment by drobtravels** [comment]
I'm not familiar with that gem, but I would advise not using as it, as it will be better learning opportunity.  Use the approach for AJAX from the end of Foundations instead.

As far as approach, instead of `enable` why not use the standard `update` action?  A single controller action can respond to multiple formats (like HTML, JS, etc).  The step you are missing before step 2 is that unchecking the box would immediately make an AJAX request with params `{enable: false}`.  

> create an enable.js.erb file that would have an onClick handler function that would trigger the enable
> method  in the controller 

No, the onClick would be in whatever view or partial shows each application.

> create an enable.js.erb file that would also hide the respective registered app when it is clicked.

Yes!

File: app/controllers/registered_apps_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    if @registered_app.save
+      flash[:notice] = "Your app was registered!"
+      redirect_to user_path(@user)
+    else
+      flash[:alert] = "There was a problem registering you app. Please make you that you give it a name and a url and try again."
+      render :new
+    end
+  end
+
+  def edit
```
Comment (drobtravels) [inline app/controllers/registered_apps_controller.rb]: no need to define this if you're not using it

File: app/views/registered_apps/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<h1>RegisteredApp#edit</h1>
+<p>Find me in app/views/registered_app/edit.html.erb</p>
```
Comment (drobtravels) [inline app/views/registered_apps/edit.html.erb]: no need for this

File: app/views/registered_apps/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<h1>RegisteredApp#index</h1>
+<p>Find me in app/views/registered_app/index.html.erb</p>
```
Comment (drobtravels) [inline app/views/registered_apps/index.html.erb]: no need for this

File: app/controllers/registered_apps_controller.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    @registered_app = RegisteredApp.new
+  end
+
+  def create
+
+    @user = User.find(params[:user_id])
+    @registered_app = RegisteredApp.new
+    @registered_app.name = params[:registered_app][:name]
+    @registered_app.url = params[:registered_app][:url]
+    @registered_app.user_id = @user.id
```
Comment (drobtravels) [inline app/controllers/registered_apps_controller.rb:16]: could you change this to something like `@registered_app = RegisteredApp.new(app_params)`?

File: app/views/registered_apps/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<div class="cointainer">
+  <h2><%=@registered_app.name%>: <%=@registered_app.url%></h2>
+</div>
```
Comment (drobtravels) [inline app/views/registered_apps/show.html.erb:3]: Is this needed for anything?

**Comment by drobtravels** [comment]
Looks pretty good.  Do you know how to fix the merge conflicts?

File: app/views/registered_apps/show.html.erb (hunk 5 above)
Comment (javierforero) [inline app/views/registered_apps/show.html.erb:3]: I thought this was required of the user story, but maybe I misinterpreted. Under build view of this check point:

> Build out the read action and views as well. The read view should display the registered application's name and URL. Lastly, construct the delete action to de-register and destroy an existing application registration. 

Was this supposed to be in the users user#show, then?

File: app/controllers/registered_apps_controller.rb (hunk 1 above)
Comment (javierforero) [inline app/controllers/registered_apps_controller.rb]: The edit method? Everything else is being used when you register an app.

File: app/controllers/registered_apps_controller.rb (hunk 4 above)
Comment (javierforero) [inline app/controllers/registered_apps_controller.rb:16]: Yes, of course.

**Comment by javierforero** [comment]
Yes, the conflict occurred because I branched off a branch (checkpoint-2) to keep working on the next checkpoint, and then I did a git pull, and somewhere in the process that created duplicate code. I ran a few of the commands github recommended and that pointed to the folder that had the conflict markers. I just deleted that line of code, staged it, and committed it. Now we're all good

**Pull request merged**